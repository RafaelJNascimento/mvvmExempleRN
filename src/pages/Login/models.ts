import React from "react";

export interface LoginViewModel {
    email: string;
    password: string;
    setEmail: React.Dispatch<React.SetStateAction<string>>;
    setPassword: React.Dispatch<React.SetStateAction<string>>;
    isLoading: boolean,
    onSubmit: () => void;
}
