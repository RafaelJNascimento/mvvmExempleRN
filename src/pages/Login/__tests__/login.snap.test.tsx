import React from 'react';
import renderer from 'react-test-renderer';

import { LoginViewModel } from "../models";
import LoginView from "../view";

jest.mock('../view.model', () => {
    const fnLoginViewModel = jest.fn();
    fnLoginViewModel.mockImplementation(
        () => ({
            email: '',
            password: '',
            isLoading: false,
            onSubmit: jest.fn(),
            setEmail: jest.fn(),
            setPassword: jest.fn(),
        } as LoginViewModel),
    );
    return fnLoginViewModel;
});

import useLoginViewModel from "../view.model";

const mockedUseLoginViewModel = useLoginViewModel as jest.Mock;

it('should render correctly', () => {
    const { toJSON } = renderer.create(<LoginView />);
    expect(toJSON()).toMatchSnapshot();
});

it('should render correctly with email', () => {
    mockedUseLoginViewModel.mockImplementation(
        () =>
        ({
            email: 'rafael@gmail.com',
            password: '',
            isLoading: false,
            onSubmit: jest.fn(),
            setEmail: jest.fn(),
            setPassword: jest.fn(),
        } as LoginViewModel),
    );

    const { toJSON } = renderer.create(<LoginView />);
    expect(toJSON()).toMatchSnapshot();
});

it('should render correctly with password', () => {
    mockedUseLoginViewModel.mockImplementation(
        () =>
        ({
            email: '',
            password: 'senhaDeTeste',
            isLoading: false,
            onSubmit: jest.fn(),
            setEmail: jest.fn(),
            setPassword: jest.fn(),
        } as LoginViewModel),
    );

    const { toJSON } = renderer.create(<LoginView />);
    expect(toJSON()).toMatchSnapshot();
});

it('should render correctly with email and password', () => {
    mockedUseLoginViewModel.mockImplementation(
        () =>
        ({
            email: 'rafael@gmail.com',
            password: 'senhaDeTeste',
            isLoading: false,
            onSubmit: jest.fn(),
            setEmail: jest.fn(),
            setPassword: jest.fn(),
        } as LoginViewModel),
    );

    const { toJSON } = renderer.create(<LoginView />);
    expect(toJSON()).toMatchSnapshot();
});

it('should render correctly while is loading', () => {
    mockedUseLoginViewModel.mockImplementation(
        () =>
        ({
            email: 'rafael@gmail.com',
            password: 'senhaDeTeste',
            isLoading: true,
            onSubmit: jest.fn(),
            setEmail: jest.fn(),
            setPassword: jest.fn(),
        } as LoginViewModel),
    );

    const { toJSON } = renderer.create(<LoginView />);
    expect(toJSON()).toMatchSnapshot();
});
