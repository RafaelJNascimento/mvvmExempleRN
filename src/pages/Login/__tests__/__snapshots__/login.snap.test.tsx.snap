// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#abbeed",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "height": "30%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="email@fleye.com.br"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value=""
    />
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="*****"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should render correctly while is loading 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#abbeed",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "height": "30%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="email@fleye.com.br"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value="rafael@gmail.com"
    />
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="*****"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value="senhaDeTeste"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
            {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should render correctly with email 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#abbeed",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "height": "30%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="email@fleye.com.br"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value="rafael@gmail.com"
    />
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="*****"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should render correctly with email and password 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#abbeed",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "height": "30%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="email@fleye.com.br"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value="rafael@gmail.com"
    />
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="*****"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value="senhaDeTeste"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should render correctly with password 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#abbeed",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "height": "30%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="email@fleye.com.br"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value=""
    />
    <Text
      style={
        {
          "color": "#424242",
          "fontSize": 15,
          "marginBottom": 5,
          "marginLeft": 2,
          "marginTop": 10,
        }
      }
    >
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="*****"
      style={
        {
          "backgroundColor": "#e5e6eb",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 10,
          "paddingLeft": 8,
          "width": "100%",
        }
      }
      value="senhaDeTeste"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
